File: src/main/java/ch/zxseitz/j3de/graphics/scene/components/IComponent.java
package ch.zxseitz.j3de.graphics.scene.components;

import ch.zxseitz.j3de.exceptions.J3deException;
import ch.zxseitz.j3de.graphics.scene.Actor;
import ch.zxseitz.j3de.math.Matrix4;

public interface IComponent {
    void update(double delta, Actor actor, Matrix4 transform) throws J3deException;
}

File: src/main/java/ch/zxseitz/j3de/graphics/scene/components/IComponent.kt
package ch.zxseitz.j3de.graphics.scene.components;

import ch.zxseitz.j3de.graphics.scene.Actor;
import ch.zxseitz.j3de.math.Matrix4;

interface IComponent {
    fun update(delta: Double, actor: Actor, transform: Matrix4);
}
